### PR TITLE
8254166: Zero: return-type warning in zeroInterpreter_zero.cpp

### PR DESCRIPTION
**Pull request by shipilev**
This breaks 11u without disabled warnings as errors, but the issue exists in head JDK as well. Might as well fix it everywhere.

Testing:
  - [x] Linux ARM32 Zero build on HEAD JDK
  - [x] Linux ARM32 Zero build on 11u

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8254166](https://bugs.openjdk.java.net/browse/JDK-8254166): Zero: return-type warning in zeroInterpreter_zero.cpp

### Reviewers
 * [Severin Gehwolf](https://openjdk.java.net/census#sgehwolf) (@jerboaa - **Reviewer**) ⚠️ Review applies to 95d624361bf32d64a4cacfb643971c74d12adefb

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/545/head:pull/545`
`$ git checkout pull/545`

### CODE REVIEW

File: src/hotspot/cpu/zero/zeroInterpreter_zero.cpp [hunk 1]
```diff
@@ -138,7 +138,9 @@ intptr_t narrow(BasicType type, intptr_t result) {
     case T_DOUBLE:
     case T_VOID:
       return result;
-    default  : ShouldNotReachHere();
+    default:
+      ShouldNotReachHere();
+      return result;
```
Comment (jerboaa) [inline src/hotspot/cpu/zero/zeroInterpreter_zero.cpp:143]: Perhaps add a `// silence compiler` comment to the return?

**Review by jerboaa — APPROVED** [review]
Looks OK to me. I'd suggest adding a comment as the return after `ShouldNotReachHere()` looks odd otherwise.

File: src/hotspot/cpu/zero/zeroInterpreter_zero.cpp (hunk 1 above)
Comment (shipilev) [reply src/hotspot/cpu/zero/zeroInterpreter_zero.cpp]: Right. Added in new revision.

**Review by jerboaa — APPROVED** [review]
Looks fine.

**Pull request closed without merging**